Source/Scene/processPbrMaterials.js

```diff
@@ -88,7 +88,7 @@ define([
         return gltf;
     }
 
-    function isPbrMaterial(material) {
+    function isMetallicRoughnessMaterial(material) {
```

Comment:
This should actually not be renamed. The function is meant to catch if PBR shaders need to be generated. It should be catching whether the material uses metallic roughness, spec gloss, or unlit. The purpose is to turn away materials that are using `KHR_techniques_webgl` or `KHR_materials_common` (1.0).

One way to test is with a material like:
```
 {
  "extensions": {
    "KHR_materials_pbrSpecularGlossiness": {}
  }
}
```

Reply:
This check probably also belongs before the `hasPbrMetallicRoughness` check:

```
        // No need to create new techniques if they already exist,
        // the shader should handle these values
        if (hasExtension(gltf, 'KHR_techniques_webgl')) {
            return gltf;
        }
```

I feel like a lot of the material existence checking was a little messy before this PR and could use some cleanup.

Reply:
Ah, good catch! I think it's helpful to have those two functions separately, so I might just add:

```
function isPbrMaterial(material){
  return isMetallicRoughnessMaterial(material) || isSpecularGlossinessMaterial(material);
}

Reply:
Sounds good - though make sure sure the extreme edge case works too. This should produce a default PBR shader.

```
"materials": [
  {
  }
]
```

Reply:
Wow, you're right. I didn't realize glTF materials were all PBR by default. That extreme case currently doesn't work in master. It should be good here now. 

So `isMetallicRoughnessMaterial()` isn't needed at all then. If any material is found, then it is metallic roughness, unless spec-gloss is also found.

Reply:
Yeah it might not be needed at all.